controllers/dscinitialization/utils.go

```diff
@@ -18,132 +19,184 @@ import (
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
 
 	dsciv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/dscinitialization/v1"
 	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster"
 	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
 	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/metadata/labels"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/resources"
 )
 
 var (
 	resourceInterval = 10 * time.Second
 	resourceTimeout  = 1 * time.Minute
 )
 
-// createOdhNamespace creates a Namespace with given name and with ODH defaults. The defaults include:
-// - Odh specific labels
-// - Pod security labels for baseline permissions
-// - ConfigMap  'odh-common-config'
-// - Network Policies 'opendatahub' that allow traffic between the ODH namespaces
-// - RoleBinding 'opendatahub'.
-func (r *DSCInitializationReconciler) createOdhNamespace(ctx context.Context, dscInit *dsciv1.DSCInitialization, name string, platform cluster.Platform) error {
+// createOperatorResource include steps:
+// - 1. validate customized application namespace || create/update default application namespace
+// - 2. patch application namespaces for managed cluster
+//   - Odh specific labels for access
+//   - Pod security labels for baseline permissions
+//
+// - 3. Patch monitoring namespace
+// - 4. Network Policies 'opendatahub' that allow traffic between the ODH namespaces
+// - 5. ConfigMap  'odh-common-config'
+// - 6. RoleBinding 'opendatahub'.
+func (r *DSCInitializationReconciler) createOperatorResource(ctx context.Context, dscInit *dsciv1.DSCInitialization, nsName string, platform cluster.Platform) error {
 	log := logf.FromContext(ctx)
-	// Expected application namespace for the given name
-	desiredNamespace := &corev1.Namespace{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: name,
-			Labels: map[string]string{
-				labels.ODH.OwnedNamespace: "true",
-				labels.SecurityEnforce:    "baseline",
-			},
-		},
-	}
 
-	// Create Application Namespace if it doesn't exist
-	foundNamespace := &corev1.Namespace{}
-	err := r.Get(ctx, client.ObjectKeyFromObject(desiredNamespace), foundNamespace)
-	if err != nil {
-		if k8serr.IsNotFound(err) {
-			log.Info("Creating namespace", "name", name)
-			// Set Controller reference
-			// err = ctrl.SetControllerReference(dscInit, desiredNamespace, r.Scheme)
-			// if err != nil {
-			//	 log.Error(err, "Unable to add OwnerReference to the Namespace")
-			//	 return err
-			// }
-			err = r.Create(ctx, desiredNamespace)
-			if err != nil && !k8serr.IsAlreadyExists(err) {
-				log.Error(err, "Unable to create namespace", "name", name)
-				return err
-			}
-		} else {
-			log.Error(err, "Unable to fetch namespace", "name", name)
+	switch nsName { // default application: create or update labels
+	case "opendatahub", "redhat-odh-applications":
```

Comment:
I fail to see the reason to have such case since `opendatahub` is a namespace name like any other, so there is no guarantees it behave differently than a user created one so if the user creates it and label it with `application-namespace: true`,  then  it should not have a label `OwnedNamespace: true`